test/Common.cpp

```diff
@@ -117,9 +117,20 @@ bool CommonOptions::parse(int argc, char const* const* argv)
 
 	po::command_line_parser cmdLineParser(argc, argv);
 	cmdLineParser.options(options);
-	po::store(cmdLineParser.run(), arguments);
+	auto parsedOptions = cmdLineParser.run();
+	po::store(parsedOptions, arguments);
 	po::notify(arguments);
 
+	for (auto const& parsedOption: parsedOptions.options)
+		if (parsedOption.position_key >= 0)
+		{
+			std::stringstream errorMessage;
+			errorMessage << "Unrecognized option: ";
+			for (auto const& token: parsedOption.original_tokens)
```

Comment:
Note to self:

From boost docs

> The original unchanged tokens this option was created from.

So this would be `--` or even `-- <followed_by_non_named_options_for_isoltest`

Reply:
This is one of the reasons why I wrote "``boost::program_options`` is weird" :-). It's basically undocumented what it "tokenizes" here, as far as I can tell... in practice the ``--`` itself is silently skipped and everything after it is considered "positional arguments", which are delimited by non-escaped spaces. Each of those seems to always contain *one* "token" in ``original_tokens``, which is the entire option string...

So e.g.:
``-- bla1 bla2`` -> two positional arguments with original_tokens being ``{ "bla1" }`` and ``{ "bla2" }`` respectively
``-- --test\ bla1 --bla2 "a b c"`` -> three positional arguments with original_tokens being ``{ "--test bla1" }``, ``{ "--bla2" }`` and ``{ "a b c" }`` respectively